x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/components/datasource_delete_provider.tsx

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, useMemo, useRef, useState } from 'react';
+import { EuiCallOut, EuiConfirmModal, EuiOverlayMask, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { useCore, sendRequest, sendDeleteDatasource } from '../../../hooks';
+import { AGENT_API_ROUTES } from '../../../../../../common/constants';
+import { AgentConfig } from '../../../../../../common/types/models';
+
+interface Props {
+  agentConfig: AgentConfig;
+  children: (deleteDatasourcePrompt: DeleteAgentConfigDatasourcePrompt) => React.ReactElement;
+}
+
+export type DeleteAgentConfigDatasourcePrompt = (
+  datasourcesToDelete: string[],
```

Comment:
I'm not sure that we plan on surfacing bulk delete of data sources anywhere in the UI so the code path for deleting multiple data sources in this file won't ever be triggered, but I'm ok with leaving the code in